### DISCUSSION TITLE
Problem with Encoder switch/Switch

### DISCUSSION THREAD

**Discussion by fp64lib** [probonopd/MiniDexed#132] [Q&A] [answered]
After checking everything twice and three-times and the encoder switch is still not working, I need some help/hints. The encoder itself works fine, reacts on every click. Pressing the encoder gives a signal, checked with the multimeter both at the switch and at GPIO26, so signal arrives through the cabling, i.e. no contact problem. To exclude internal errors of the switch, I even replaced it by using a simple push button switch, which  per default pulled high to 5V (like the encoder witch) and when pressed will bring down the signal line. Also this does not work. Any hints?

Relevant part of minidexed.ini:
```
# KY-040 Rotary Encoder
EncoderEnabled=1
EncoderPinClock=5
EncoderPinData=6
EncoderPinSwitch=26
```

My hardware configuration: 
Raspberry 2B+ --> GPIO 26 is physical header pin 37 (second last pin on "inner " side towards CPU, two gaps between this pin and GPIO 5&6 which are used for the encoder, which works)

**Comment by probonopd** [comment]
Are you 100% sure your Raspberry Pi pin is not damaged?

On my rotary encoder (HW-040) the SW and GND contacts are normally open and get closed (shorted) when the switch is pressed. On the back of my encoder module, R1 is unpopulated and R2, R3 are populated with 10k resistors. It came that way from the factory. I have not attached any additional pullups.

hope this helps!

**Comment by fp64lib** [comment, marked as answer]
All pins and components were fine, problem was sitting in front of it. Did'nt expect, that you have to double click to enter a menu item and click for exit - expected it the other way. All is working fine. Great sound!

**Reply by craigyjp** [reply]
I fell for the same thing, had to read the manual, I find it quite annoying to get used to the double click and end up leaving the menu as I click only once.

**Reply by probonopd** [reply]
Double click = open things. Like on the computer.

Do you know any devices where you double click to _leave_ something?
Just curious.

**Reply by craigyjp** [reply]
No, double click to leave isn't a thing, I've only just built the device so getting used to the menus was my only issue. By the way I used an OLED for the display and that works fine.

**Reply by probonopd** [reply]
Which OLED exactly? I'd like to put it in the wiki.

**Reply by craigyjp** [reply]
MIDAS  MCOB21605C1V-EWP

Standard 1602 character display, but I would imagine most 1602 OLEDs would work.![20220421_203617.jpg](https://user-images.githubusercontent.com/51026305/164548020-8135dfcd-7a5e-47af-ab06-a5cef2ae7070.jpg)

**Reply by probonopd** [reply]
Wow, nice setup :+1: although a bit overkill to use a RPi 400 ;-)

**Reply by craigyjp** [reply]
All I could get here in the Ukraine quickly, really wanted a board and have no experience of the Pi's, thought I was getting an eval kit with a keyboard. A bit surprised by what I received, can I increase the amount of instruments?
